sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -999,6 +1041,35 @@ static String getExtractDestinationUri(String extractDestinationDir) {
       return toBuilder().setValidate(false).build();
     }
 
+    /** Choose the method used to write data to BigQuery. See the Javadoc on {@link Method} for
+     * information and restrictions of the different methods.
+     */
+    public Write<T> withMethod(Method method) {
+      return toBuilder().setMethod(method).build();
+    }
+
+    /** Choose the frequency at which file writes are triggered.
+     *
+     * <p>This is only applicable when the write method is set to {@link Method#FILE_LOADS}.
+     * Every triggeringFrequency duration, a BigQuery load job will be generated for all
+     * the files written in the last pane. BigQuery has limits on how many load jobs can be
+     * triggered per day, so be careful not to set this duration too low, or you may exceed daily
+     * quota. Often this is set to 5 or 10 minutes to ensure that the project stays well under the
+     * BigQuery quota. See https://cloud.google.com/bigquery/quota-policy for more information about
+     * BigQuery quotas.
+     */
+    public Write<T> withTriggeringFrequency(Duration triggeringFrequency) {
+      return toBuilder().setTriggeringFrequency(triggeringFrequency).build();
+    }
+
+    /**
+     * Control how many file shards are written when using BigQuery load jobs. Currently this is
```

Comment:
Yes, there's no obvious default value. We could have runners choose a default runner here (e.g. based on # workers like the DataflowRunner does for WriteFiles in streaming).

Yes, it's allowed to set this for bounded input collection, though not encouraged (we would prefer user to stick to dynamically-chosen sharding when possible).